### PR TITLE
Allow adding audio clips from a local file - some improvements

### PR DESCRIPTION
**Pull request by johndem**
I looked into hssm's [suggestions](https://github.com/ankidroid/Anki-Android/pull/4050) and made some improvements to the code and UI.
As for **Audio clip is using the microphone icon (the same as audio recording). We should probably make this something else.** I didn't find an appropriate png file in the project's assets to use so I leave it up to you to decide.

### CODE REVIEW

**Comment by arooooon** [comment]
You can use one of these icons and then add it to the project:
https://design.google.com/icons/

**Comment by hssm** [comment]
I'll test it out later today.

@arooooon Thanks for the link. [Library music](https://design.google.com/icons/#ic_library_music) seems to be a good fit.

**Comment by timrae** [comment]
It looks like my two suggestions slipped through the cracks. It's probably faster for me to make make the edits myself (I'll add the icon too) and submit a modified PR

File: AnkiDroid/src/main/java/com/ichi2/anki/multimediacard/fields/AudioClipField.java [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+//    @Override
+//    public String getText() {
+//        return null;
+//    }
+//
+//
+//    @Override
+//    public boolean setText(String text) {
+//        return false;
+//    }
```
Comment (hssm) [inline AnkiDroid/src/main/java/com/ichi2/anki/multimediacard/fields/AudioClipField.java]: There's no need to leave large blocks of commented code.

Comment (johndem) [inline AnkiDroid/src/main/java/com/ichi2/anki/multimediacard/fields/AudioClipField.java]: Sorry it seems I forgot to remove them in the process. :P

**Comment by sibilek** [comment]
I am so happy for this enhancement. Adding local audio file is a great idea. Thank you. 
I have an problem which is related to audio adding from local file. 

I have Colombian friend who is learning Turkish and also helping me with Spanish. We are talking by WhatsApp and sometimes he teaches me new Spanish words and also sending me pronounciations by WhatsApp voice note. Earlier WhatsApp was using .aac file format which is supported by AnkiDroid but I think WhatsApp changed voice note file to .opus file which AnkiDroid doesn't support. 

Is it possible to add .opus file support to AnkiDroid? Or if not possible;
Within this "add local audio file enhencment", is it possible to convert and add .opus file as mp3 in AnkiDroid? 

Many thanks for your contributions. 


P.S: on playstore there are free app which converts .opus to mp3 but it just prolongs the adding audio process to AnkiDroid. Here is the app link
https://play.google.com/store/apps/details?id=grant.opus.to.mp3

**Comment by sibilek** [comment]
I want to mention one more thing. There are 2 useful app which contribute to AnkiDroid by means of adding audio clips.

One of them is [Sound4AnkiDroid](https://play.google.com/store/apps/details?id=com.learnerstechlab.soundcapture4ankidroid&hl=en
). He is so kind. Even once I asked an TTS sound export enhencment for the app he implement it. it's an opensource app. Here is [GitHub link](https://github.com/cojcoj/android-soundcapture4ankidroid). 
it still works pretty good. But only adds English words sound files from various dictionaries. and it can add tts sound file (for a word or for a long sentence) in any tts lang available in phone

The other one is [AB Repeat Player](http://apk-dl.com/ab-repeat-player). This app is a audio/video player for language learners. The app also does a lot but for our stiuation most importantly, it exports selected sound clips from audio to AnkiDroid.

 Both of them developed by same person. Maybe if we can get in touch with him, he would like to contribute to AnkiDroid by means of audio support.

Have a nice day

**Comment by mikehardy** [comment]
Looks like this is close but still unmerged.

For completeness, @timrae mentions he had a suggestion that was orphaned on the pull request related to this work prior -

use ACTION_GET_CONTENT in place of ACTION_PICK - https://github.com/ankidroid/Anki-Android/pull/4050#discussion_r49688944

If there is still interest in this feature I may be able to pick it up and shepherd it to a merge. I haven't worked in Android UI yet and this is an easier intro than profile support as a first attempt

**Comment by mikehardy** [comment]
#4756 will carry this work into 2.9, thanks @johndem

**Pull request closed without merging**